src/reporting/market.se

```diff
@@ -146,18 +144,18 @@ def disputeAutomatedReport():
     # intentionally does not migrate the market as automated report markets won't actually migrate unless a dispute bond has been placed or the automated report doesn't occur
     require(not self.isFinalized())
     require(self.isInAutomatedDisputePhase())
-    self.getReputationToken().trustedTransfer(msg.sender, self, AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT)
-    self.automatedReporterDisputeBondHolder = msg.sender
+    require(not self.automatedReporterDisputeBondToken)
+    self.automatedReporterDisputeBondToken = DISPUTE_BOND_TOKEN_FACTORY.createDisputeBondToken(self.controller, self, msg.sender, AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT, self.tentativeWinningPayoutDistributionHash)
+    self.transferBondAmountToDisputeBondToken(msg.sender, self.automatedReporterDisputeBondToken, AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT)
     self.reportingWindow.updateMarketPhase()
     return 1
 
 def disputeLimitedReporters():
     assertNoValue()
     self.migrateThroughAllForks()
-    require(self.isInLimitedDisutePhase())
-    self.getReputationToken().trustedTransfer(msg.sender, self, LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT)
-    self.limitedReportersDisputeBondHolder = msg.sender
-    self.limitedReportersPayoutDistributionHash = self.tentativeWinningPayoutDistributionHash
+    require(self.isInLimitedDisputePhase())
```

Comment:
For this and `disputeAllReporters` shouldn't we also require that there is not already a dispute bond for this phase placed as we do for the automated dispute?